sphinxgallery/gen_rst.py

```diff
@@ -292,157 +254,175 @@ def scale_image(in_fname, out_fname, max_width, max_height):
-
-                fig.savefig(image_path % fig_mngr.num, **kwargs)
-                figure_list.append(image_fname % fig_mngr.num)
-        except:
-            print(80 * '_')
-            print('%s is not compiling:' % fname)
-            traceback.print_exc()
-            print(80 * '_')
-        finally:
-            os.chdir(cwd)
-            sys.stdout = orig_stdout
-
-        print(" - time elapsed : %.2g sec" % time_elapsed)
-    else:
-        figure_list = [f[len(image_dir):]
-                       for f in glob.glob(image_path.replace("%03d",
-                                            '[0-9][0-9][0-9]'))]
-    figure_list.sort()
+    thumb_file = os.path.join(thumb_dir,
+                              'sphx_glr_%s_thumb.png' % base_image_name)
 
-    # generate thumb file
     if os.path.exists(first_image_file):
         scale_image(first_image_file, thumb_file, 400, 280)
+    elif not os.path.exists(thumb_file):
+        # create something to replace the thumbnail
+        scale_image(os.path.join(glr_path_static(), 'no_image.png'),
+                    thumb_file, 200, 140)
 
-    # Depending on whether we have one or more figures, we're using a
-    # horizontal list or a single rst call to 'image'.
-    if len(figure_list) == 1:
-        figure_name = figure_list[0]
-        image_list = SINGLE_IMAGE % figure_name.lstrip('/')
-    else:
-        image_list = HLIST_HEADER
-        for figure_name in figure_list:
-            image_list += HLIST_IMAGE_TEMPLATE % figure_name.lstrip('/')
 
-    return image_list, time_elapsed, stdout
+def generate_dir_rst(src_dir, target_dir, gallery_conf, seen_backrefs):
+    """Generate the gallery reStructuredText for an example directory"""
+    if not os.path.exists(os.path.join(src_dir, 'README.txt')):
+        print(80 * '_')
+        print('Example directory %s does not have a README.txt file' %
+              src_dir)
+        print('Skipping this directory')
+        print(80 * '_')
+        return ""  # because string is an expected return type
+
+    fhindex = open(os.path.join(src_dir, 'README.txt')).read()
+    if not os.path.exists(target_dir):
+        os.makedirs(target_dir)
+    sorted_listdir = [fname for fname in sorted(os.listdir(src_dir))
+                                if fname.endswith('py')]
+    for fname in sorted_listdir:
+        generate_file_rst(fname, target_dir, src_dir)
+        new_fname = os.path.join(src_dir, fname)
+        intro = extract_intro(new_fname)
+        write_backreferences(seen_backrefs, gallery_conf,
+                             target_dir, fname, intro)
+
+        fhindex += _thumbnail_div(target_dir, fname, intro)
+        fhindex += """
+
+.. toctree::
+   :hidden:
+
+   /%s/%s\n""" % (target_dir, fname[:-3])
+
+
+    # clear at the end of the section
+    fhindex += """.. raw:: html\n
+    <div style='clear:both'></div>\n\n"""
+
+    return fhindex
+
+
+def execute_script(code_block, example_globals, image_path, fig_count, src_file):
```

Comment:
PEP8 line too long.

Pro tip: take some time setting up PEP8 checks in your editor of choice, it will be worth the investment in the long run.